src/components/BulkMemberManager/hooks/useResults/useResults.js

```diff
@@ -1,16 +1,24 @@
 import { useDataQuery } from '@dhis2/app-runtime'
-import { useState, useEffect, useCallback, useRef } from 'react'
+import { useState, useEffect, useCallback, useRef, useMemo } from 'react'
 import { useDebounce } from 'use-debounce'
 import { useSet } from './useSet.js'
 
 export const useResults = ({
+    pendingChanges,
     canManageMembers,
     allQuery,
     membersGistQuery,
     transformQueryResponse,
     filterDebounceMs,
     mode,
 }) => {
+    const pendingChangesForMode = useMemo(() => {
```

Comment:
is there a particular reason to use useMemo here? It seems like we don't need a stable reference, and I'm not sure if the calculation is particularly expensive? It seems like the cost of making the comparison of pendingChanges might outweigh the benefit of the memoization? (I might have missed something though 😅 and also think either memoizing or not memoizing is okay 🙃)

Reply:
I agree there's probably no need to do this. Have removed the memoization.